aten/src/ATen/native/TensorIterator.h

```diff
@@ -96,6 +96,10 @@ struct CAFFE2_API OperandInfo {
   // (e.g. if dtype is changed)
   Tensor original_tensor;
 
+  // If a tensor is both an input and an output, this lets the input
+  //   operand point to the output operand for a quick lookup
+  OperandInfo* corresponding_output = nullptr;
```

Comment:
Is this an aliasing relation (if so, what if an input is multiple outputs) or a more technical definition (e.g., the self input in an inplace operation, which is both an input/output)?

Reply:
I also don't like this raw pointer. I'll get rid of it and fold the aliasing check into the logic which reuses the temporary/ies.

Reply:
I removed this optimization to avoid complicating the code and dragging down the common path with additional operand enumerations.